packages/devtools-launchpad/src/components/LandingPage.js

```diff
@@ -134,30 +136,36 @@ const LandingPage = React.createClass({
     const isConnected = name == "Firefox"
       ? this.state.firefoxConnected
       : this.state.chromeConnected;
-    const launchButton = isConnected ? null : dom.input({
+
+    const connectedStateText = isNodeSelected ?
+      null : `Please open a tab in ${name}`;
+
+    const launchButton = isConnected ? connectedStateText : dom.input({
       type: "button",
       value: `Launch ${selectedPane}`,
       onClick: () => launchBrowser(selectedPane)
     });
+
+    const targetsContent = currentTargetsExist ? dom.header({},
+      dom.input({
+        ref: "filterInput",
+        placeholder: "Filter tabs",
+        value: filterString,
+        autoFocus: true,
+        type: "search",
+        onChange: e => this.onFilterChange(e.target.value),
+        onKeyDown: e => {
+          if (targets.size === 1 && e.keyCode === 13) {
+            this.onTabClick(targets.first(), paramName);
+          }
+        }
+      })
+    ) : dom.div({ className: "hero" }, launchButton);
```

Comment:
![launchpad](https://cloud.githubusercontent.com/assets/1637928/26036380/440ef18c-38e5-11e7-9226-648a2308d5cb.gif)
Here's how it looks like. Few thoughts
1. There's an issue #370 with detecting firefox connected state after launching a browser (I'll open a separate PR on it)
2. Can ES6 syntax for React be used?
3. Node - 👍